sktime/container/array.py

```diff
@@ -0,0 +1,850 @@
+        object of np.ndarrays in which the first entry will be interpreted as
+        values and the second entry as indices.
+
+    Returns
+    -------
+    TimeArray
+    """
+    n = len(data)
+
+    out_d = []
+    out_i = []
+    widths = np.full((n, ), np.nan, dtype=np.float)
+
+    for idx in range(n):
+        ts = data[idx]
+
+        # TODO: think about other constructors that might make sense
+        if isinstance(ts, (TimeBase, TimeArray)):
+            check_data_index(ts.data, ts.time_index)
+            out_d.append(ts.data)
+            out_i.append(ts.time_index)
+            widths[idx] = ts.data.shape[1]
+        elif isinstance(ts, (tuple, list, np.ndarray)):
+            out_d.append(ensure_2d(np.array(ts[0])))
+            out_i.append(ensure_2d(np.array(ts[1])))
+            widths[idx] = out_d[idx].shape[1]
+        elif isinstance(ts, pd.Series):
+            out_d.append(ts.values)
+            out_i.append(ts.index.values)
+            widths[idx] = ts.values.size
+        elif ts is None:
+            out_d.append(None)
+            out_i.append(None)
+        else:
+            raise TypeError("Input must be valid timeseries objects: {ts}")
+
+    # Deal with missing
+    mask = np.isnan(widths)
+    if np.all(mask):
+        # All list elements represented missing timeseries
+        return TimeArray(empty((n, 0)))
+    elif np.any(mask):
+        # Some list elements represented missing timeseries
+        # TODO: is there a better way to do this?
+        w = out_d[int(np.where(~mask)[0][0])].shape[1]
+        out_d = [np.full((1, w), np.nan, np.float) if m else d
+                 for m, d in zip(mask, out_d)]
+        out_i = [np.full((1, w), np.nan, np.float) if m else i
+                 for m, i in zip(mask, out_i)]
+
+    # Ensure that all the widths are equal
+    if not np.all(widths[~mask] == widths[~mask][0]):
+        raise ValueError("The width of each timeseries must be the equal.")
+
+    return TimeArray(np.vstack(out_d), np.vstack(out_i))
+
+
+def from_pandas(data):
+    """
+    Create TimeArray object from a nested pd.Series or from a pd.DataFrame
+
+    If data is a nested pd.Series, each sub-series will be interpreted as a row
+    in the TimeArray. If data is a pd.Dataframe, each row in the pd.Dataframe
+    will be interpreted as a row in the TimeArray and the column index will be
+    used as the index.
+
+    Parameters
+    ----------
+    data : pd.Series, pd.DataFrame
+
+    Returns
+    -------
+    TimeArray
+    """
+    if isinstance(data, pd.Series):
+        return from_list(data)
+    elif isinstance(data, pd.DataFrame):
+        time_index = data.columns.values[np.newaxis, :]
+        time_index = np.repeat(time_index, data.shape[0], axis=0)
+        return TimeArray(data.to_numpy(), time_index)
+    else:
+        raise TypeError("Input must be a pandas Series or DataFrame: {data}")
+
+
+def from_ts(ts):
```

Comment:
what do you mean by `ts`? Could you add some reference to the doctoring?

Reply:
`ts` was meant to be the ts data format as used in the loading utilities [here](https://github.com/alan-turing-institute/sktime/blob/master/sktime/utils/load_data.py). 

Ultimately, we would provide a same range of importing utilities as we do currently. I needed one that allows me to factorise an object (i.e. turn it into a string) and the ts-like format seemed to be the simplest way to do that. `from_ts`/`to_ts` is therefore a very rudimentary implementation of such a loading function for the sole purpose of factorisation. I can add this to the docstring.